README.md

```diff
@@ -282,3 +282,11 @@ Deleted posts are restored automatically before the push occurs. However, you ca
 ```bash
 npm run dev-restore
 ```
+
+### Optimizing Images
+
+The pre-push workflow will validate image size and fail if there are images bigger than 400kb in size. It will also warn you about using PNG images as they are lossless and occupy more space, for a blog JPEGs or WebP images are better suited. So do consider using those formats. And ideally images should be under 1200px in width as there is no reason to use larger resolution since the blog content is rendered in a 900px width container.
+
+You can use `npm run optimize-images` to optimize all images in the `_source/_assets/img` directory. This will resize and compress the images and write them to new files. So make sure to update the usage and delete the original file.
```

Comment:
Why does it write a new file versus overwriting the old one? Git diff should allow you to see the difference.

Reply:
I can make that the default if you prefer that. I didn't do that mainly since this will create a JPG and replacing will use PNG if it was originally that and hence will cause a PNG warning. I thought since this will happen per post per author, the author can just decide what to do

Reply:
In that case, I think it's OK to keep it as-is.